prowler/providers/aws/aws_provider.py

```diff
@@ -607,20 +613,22 @@ def setup_assumed_session(
             # that needs to be a method without arguments that retrieves a new set of fresh credentials
             # assuming the role again.
             assumed_refreshable_credentials = RefreshableCredentials(
-                access_key=assumed_role_credentials.aws_access_key_id,
-                secret_key=assumed_role_credentials.aws_secret_access_key,
-                token=assumed_role_credentials.aws_session_token,
-                expiry_time=assumed_role_credentials.expiration,
-                refresh_using=self.refresh_credentials,
+                access_key=assumed_role_configuration.credentials.aws_access_key_id,
+                secret_key=assumed_role_configuration.credentials.aws_secret_access_key,
+                token=assumed_role_configuration.credentials.aws_session_token,
+                expiry_time=assumed_role_configuration.credentials.expiration,
+                refresh_using=lambda: AwsProvider.refresh_credentials(
+                    assumed_role_configuration, session
+                ),
```

Comment:
Why can't we do this?
```suggestion
                refresh_using=lambda: AwsProvider.refresh_credentials,
```

Reply:
I think we can't, @pedrooot @MrCloudSec what do you think?

Reply:
Nope, we can't since the refresh credentials method is now static and we need to pass the `assumed_role_configuration` and `session`

Reply:
Great catch, I totally skipped it.